tap_workday_raas/discover.py

```diff
@@ -8,39 +8,94 @@
 
 LOGGER = singer.get_logger()
 
-def _type_to_schema(elem_type):
+def _element_to_schema(element):
+    elem_type = element.attrib['type'].split(':')[1]
+    is_nullable = element.attrib.get('minOccurs') == '0'
+    # NB: The only actual value I've seen for maxOccurs is "unbounded". There may be other possible values
+    is_list = element.attrib.get("maxOccurs") == 'unbounded'
+
+    schema = {}
+
     if elem_type in ('date', 'dateTime'):
-        return {
-            'type': ['string', 'null'],
+        schema = {
+            'type': ['string'],
             'format': 'date-time'
         }
     elif elem_type == 'decimal':
         #TODO What else do we need for decimal types?
-        return {
-            'type': ['number', 'null']
+        schema = {
+            'type': ['number'],
+            'format': 'decimal'
```

Comment:
Maybe not just yet if this tap uses the transformer, but `"type": "string", "format": "singer-decimal"` would be good.

Reply:
Done, but the transformer treats it as a number, ignoring the format.